# 2.6.33 deleted the Hayes esp driver, so the ioctls to get and set its
# configuration are no longer needed, and they break setserial by making it
# believe that it should compile in the code that supports this modem.

2010/06/16
Created-by: Jeff Dike <jdike@linux.intel.com>

# remove additional IOCTL defines that were in arch specific headers

2010/09/02
Created-by: Saul Wold <saul.wold@intel.com>

Index: linux-2.6.34/include/asm-generic/ioctls.h
===================================================================
--- linux-2.6.34.orig/include/asm-generic/ioctls.h
+++ linux-2.6.34/include/asm-generic/ioctls.h
@@ -86,16 +86,6 @@
 #define TIOCMIWAIT	0x545C	/* wait for a change on serial input line(s) */
 #define TIOCGICOUNT	0x545D	/* read serial port inline interrupt counts */
 
-/*
- * some architectures define FIOQSIZE as 0x545E, which is used for
- * TIOCGHAYESESP on others
- */
-#ifndef FIOQSIZE
-# define TIOCGHAYESESP	0x545E  /* Get Hayes ESP configuration */
-# define TIOCSHAYESESP	0x545F  /* Set Hayes ESP configuration */
-# define FIOQSIZE	0x5460
-#endif
-
 /* Used for packet mode */
 #define TIOCPKT_DATA		 0
 #define TIOCPKT_FLUSHREAD	 1

Index: linux-2.6.34/include/asm-generic/ioctls.h
===================================================================
--- linux-2.6.34/arch/mips/include/asm/ioctls.h	2010-05-16 14:17:36.000000000 -0700
+++ linux-2.6.34_fixed/arch/mips/include/asm/ioctls.h	2010-09-02 14:54:14.641763250 -0700
@@ -103,7 +103,5 @@
 #define TIOCSERSETMULTI 0x5490 /* Set multiport config */
 #define TIOCMIWAIT      0x5491 /* wait for a change on serial input line(s) */
 #define TIOCGICOUNT     0x5492 /* read serial port inline interrupt counts */
-#define TIOCGHAYESESP	0x5493 /* Get Hayes ESP configuration */
-#define TIOCSHAYESESP	0x5494 /* Set Hayes ESP configuration */
 
 #endif /* __ASM_IOCTLS_H */
